src/features/DisplayControls/DrugChart/tests/useFetchMedications.spec.js

```diff
@@ -0,0 +1,40 @@
+import { renderHook } from "@testing-library/react-hooks";
+import { useFetchMedications } from "../hooks/useFetchMedications";
+import { testData } from "./useFetchMedicationMockData";
+
+const mockFetchMedications = jest.fn();
+
+jest.mock("../utils/DrugChartUtils", () => ({
+  fetchMedications: () => mockFetchMedications(),
+}));
+
+afterEach(() => {
+  jest.resetAllMocks();
+});
+
+beforeEach(() => {
+  mockFetchMedications.mockResolvedValue({
+    data: testData,
+  });
+});
+
+describe("useFetchMedications", () => {
+  it("should return medications", async () => {
+    const { result, waitForNextUpdate } = renderHook(() =>
+      useFetchMedications()
+    );
+    await waitForNextUpdate();
+
+    expect(result.current.drugChartData).toEqual(testData);
+  });
+
+  it("should return loading state", async () => {
+    const { result, waitForNextUpdate } = renderHook(() =>
+      useFetchMedications()
+    );
+
+    expect(result.current.isLoading).toEqual(true);
+    await waitForNextUpdate();
+    expect(result.current.isLoading).toEqual(false);
+  });
+});
```

Comment:
can there be a test to assert on failure senario?

Reply:
added